### PR TITLE
Add lighthouse steps for Azure and RHEL

### PR DESCRIPTION
**Pull request by fhlavac**
This adds back reverted https://github.com/RedHatInsights/sources-ui/pull/919

https://issues.redhat.com/browse/RHCLOUD-18688

### CODE REVIEW

**Comment by Hyperkid123** [comment]
@lindgrenj6 are we positive we can now add back the steps? Is not going to create the same issues as it did the first time?

**Comment by lindgrenj6** [comment]
We _only_ want this in the stage environment so the app team can test it.

So basically until this is prod-ready we'll have to cherry-pick commits to promote them to prod, luckily not a lot of changes happen!

**Comment by Hyperkid123** [comment]
@fhlavac can you add a condition to only enable this in stage? We don't want to block other deployments. Possibly use feature flags and load them during the init process.

File: src/test/__mocks__/@unleash/proxy-client-react.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+const lib = jest.requireActual('@unleash/proxy-client-react');
+
+const pf = {
+  ...lib,
+  useFlag: jest.fn(() => false),
+};
+
+module.exports = pf;
```
Comment (Hyperkid123) [inline src/test/__mocks__/@unleash/proxy-client-react.js:8]: This is not PF :wink:

**Review by Hyperkid123 — COMMENTED** [review]
Just a little nitpick

**Review by Hyperkid123 — APPROVED** [review]

**Pull request merged**